### PR TITLE
Improve log readability

### PR DESCRIPTION
**Pull request by angelcar**
### Summary

This PR is to improve the readability of ECS agent logs. The code changes don't include any major changes in logic, only changes to logging.
### Implementation details

Expanded the use of structured logger to improve most relevant logs. In order to decrease verbosity, now we log ony task id instead of the full task arn, which should be sufficient.
### Testing

* Unit tests
* Integ tests

New tests cover the changes: yes
### Description for the changelog

* Enhancement - Improve log readability
### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by singholt — COMMENTED** [review]
LGTM overall, just some nit picks and questions.

File: agent/api/ecsclient/client.go [hunk 1]
```diff
@@ -587,8 +589,11 @@ func (client *APIECSClient) discoverPollEndpoint(containerInstanceArn string) (*
 	if !expired && found {
 		// Cache hit and not expired. Return the output.
 		if output, ok := cachedEndpoint.(*ecs.DiscoverPollEndpointOutput); ok {
-			seelog.Infof("Using cached DiscoverPollEndpoint. endpoint=%s telemetryEndpoint=%s containerInstanceARN=%s",
```
[the remark this answers is not in the document]
Comment (angelcar) [reply agent/api/ecsclient/client.go:590]: Yep, we can change logs incrementally. For now I changed the ones I think make it very difficult to debug stuff. I'd like to keep this PR "minimal"

File: agent/api/eni/eniattachment.go [hunk 2]
```diff
@@ -55,6 +58,25 @@ type ENIAttachment struct {
 	guard sync.RWMutex
 }
 
+func getEniAttachmentLogFields(eni *ENIAttachment, duration time.Duration) logger.Fields {
+	fields := logger.Fields{
```
[the remark this answers is not in the document]
Comment (angelcar) [reply agent/api/eni/eniattachment.go:62]: CNI plugins live in a different repo. We could include the them later. For now I'm mostly interested in agent logs.

File: agent/api/task/task.go [hunk 3]
```diff
@@ -1021,7 +1067,10 @@ func (task *Task) addFirelensContainerDependency() error {
 	if firelensContainer.HasContainerDependencies() {
 		// If firelens container has any container dependency, we don't add internal container dependency that depends
 		// on it in order to be safe (otherwise we need to deal with circular dependency).
-		seelog.Warnf("Not adding container dependency to let firelens container %s start first, because it has dependency on other containers.", firelensContainer.Name)
+		logger.Warn("Not adding container dependency to let firelens container start first sunce it has dependency on other containers.", logger.Fields{
```
[the remark this answers is not in the document]
Comment (angelcar) [reply agent/api/task/task.go:1070]: nice catch! will fix!

File: agent/eni/watcher/watcher_linux.go [hunk 4]
```diff
@@ -109,7 +112,9 @@ func (eniWatcher *ENIWatcher) reconcileOnce(withRetry bool) error {
 		if withRetry {
 			go func(ctx context.Context, macAddress string, timeout time.Duration) {
 				if err := eniWatcher.sendENIStateChangeWithRetries(ctx, macAddress, timeout); err != nil {
-					log.Infof("ENI watcher event-handler: unable to send state change: %v", err)
+					logger.Info("Unable to send state ENI change", logger.Fields{
```
[the remark this answers is not in the document]
Comment (angelcar) [reply agent/eni/watcher/watcher_linux.go:115]: Yeah. Plus I changed log level to error

File: agent/api/task/task.go [hunk 5]
```diff
@@ -1883,9 +1970,13 @@ func (task *Task) dockerHostBinds(container *apicontainer.Container) ([]string,
 		}
 
 		if hv.Source() == "" || mountPoint.ContainerPath == "" {
-			seelog.Errorf(
-				"Unable to resolve volume mounts for container [%s]; invalid path: [%s]; [%s] -> [%s] in task: [%s]",
-				container.Name, mountPoint.SourceVolume, hv.Source(), mountPoint.ContainerPath, task.String())
+			logger.Error("Unable to resolve volume mounts for container; invalid path", logger.Fields{
+				field.TaskID:    task.GetID(),
+				field.Container: container.Name,
+				field.Volume:    hv.Source(),
```
[the remark this answers is not in the document]
Comment (angelcar) [reply agent/api/task/task.go:1976]: oops, yep, they're backwards. Thx.

**Review by singholt — APPROVED** [review]
🚀

**Review by sparrc — APPROVED** [review]

**Review by ubhattacharjya — APPROVED** [review]

**Review by YashdalfTheGray — APPROVED** [review]

**Pull request merged**